Dpf

codyrancher opened the pull request:
<!-- This template is for Devs to give QA details before moving the issue To-Test -->
### Summary
Fixes #
<!-- Define findings related to the feature or bug issue. -->

### Occurred changes and/or fixed issues
<!-- Include information of the changes, including collateral areas which have been affected by this PR as requirement or for convenience. -->

### Technical notes summary
<!-- Outline technical changes which may pass unobserved or may help to understand the process of solving the issue -->

### Areas or cases that should be tested
<!-- Areas that should be tested can include Airgap checks, Rancher upgrades, K8s upgrade, etc. -->
<!-- Which browser did you use for local testing? The reviewer should test with a different browser. -->
<!-- Add missing steps or rewrite them if have been missed or to complement existing information. This should define a clear way to reproduce it and not an approximation. -->

### Areas which could experience regressions
<!-- Create a detailed list of areas to be analyzed which may be affected by the changes, which would require a prior research to avoid regressions. -->

### Screenshot/Video
<!-- Attach screenshot or video of the changes and eventual comparison if you find it necessary -->

### Checklist
- [ ] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [ ] The PR has a Milestone <!-- The milestone should automatically be assigned if the linked issue has one, but does not always happen (incorrectly linked, issue has no milestone, etc) -->
- [ ] The PR template has been filled out
- [ ] The PR has been self reviewed <!-- There are no TODOs, no incorrect files in the PR, all the required files are there, no commented out code, etc-->
- [ ] The PR has a reviewer assigned
- [ ] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [ ] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes
- [ ] The PR has been reviewed in terms of Accessibility

codyrancher commented on shell/pages/explorer/resource/detail/configmap.vue:
```diff
@@ -0,0 +1,27 @@
+
+const store = useStore();
+
+const fetch = useFetch(async() => {
```
I was originally using async setup blocks with `<Suspense>` but this was causing one of our e2e tests to fail in CI but not locally. I eventually moved around this by adding the fetch composable.

rak-phillip commented on shell/components/ResourceDetail/index.vue:
```diff
@@ -1,569 +1,70 @@
+  // 'batch.cronjob':     require('@shell/pages/explorer/resource/detail/batch.cronjob.vue').default,
+  // 'batch.job':         require('@shell/pages/explorer/resource/detail/batch.job.vue').default,
+  // 'cluster-dashboard': require('@shell/pages/explorer/resource/detail/cluster-dashboard.vue').default,
+  configmap: require('@shell/pages/explorer/resource/detail/configmap.vue').default,
```
optional: Consider if using dynamic imports instead of `require` can fit for this use case. Previous investigations have shown that `require` behaves in such a way that webpack is unable to split code dynamically, leading to larger bundle sizes.

#12246

rak-phillip commented on shell/components/ResourceDetail/index.vue:
```diff
@@ -1,569 +1,70 @@
-      this.configureResource(userId, resource);
-    }
-  }
+// Ideally I'd prefer to have separate routes/pages for each of these but our app makes a
+// fair amount of assumptions around having one detail page for each resource and that the
+// detail, config, edit, yaml, create pages are all derived from the same page.
```
I think this is a fair tradeoff that preserves our existing routing and structure and better exposes which resources will be displayed and how. IMO this approach will be easier to debug in the future.

rak-phillip reviewed (approved):
LGTM - I'm happy to have an in-depth discussion on loading ResourceDetail to review tradeoffs of different approaches, but I like the approach that's on display here. 

I left a few optional comments to consider before merging.

rak-phillip commented on shell/components/Resource/Detail/Page.vue:
```diff
@@ -0,0 +1,22 @@
+    </div>
+    <div
+      v-if="$slots['middle-area']"
+      class="middle-area mmt-6"
```
What is `mmt-6`? Are we missing styles? for the page at this time?

codyrancher commented:
> I'm happy to have an in-depth discussion on loading ResourceDetail to review tradeoffs of different approaches, but I like the approach that's on display here.

We may have to, this has  been a big headache. 

I've changed where we use suspense, disabled suspensible on a number of components, moved off of async setup() and tried to create a useFetch() composable. This one has been the least problematic but there may be a better solution out there.

rak-phillip reviewed (approved):
LGTM
